lib/core/src/Cardano/Wallet/Primitive/CoinSelection/MA/RoundRobin.hs

```diff
@@ -130,23 +130,34 @@ runSelection available minimumBalance =
     assetSelectionLens (asset, minimumAssetQuantity) = SelectionLens
         { currentQuantity = assetQuantity asset . selected
         , minimumQuantity = unTokenQuantity minimumAssetQuantity
-        , selectQuantity = selectMatchingQuantity $ WithAsset asset
+        , selectQuantity = selectMatchingQuantity
+            [ WithAsset asset
+            ]
         }
 
     coinSelectionLens :: SelectionLens m SelectionState
     coinSelectionLens = SelectionLens
         { currentQuantity = coinQuantity . selected
         , minimumQuantity = fromIntegral $ unCoin minimumCoinQuantity
-        , selectQuantity = selectMatchingQuantity Any
+        , selectQuantity = selectMatchingQuantity
+            [ WithAdaOnly
+            , Any
```

Comment:
👍🏻 

I also thought of the same idea earlier on, but ran out of time before adding it.

I think it's a nice improvement, as it reduces the chance of "accidentally" selecting an output with some asset we're not actually interested in.

I suspect we could do the same for other assets: preferentially select inputs with only that asset (and no others), before moving on to selecting inputs that contain that asset (but may also contain other assets).

Reply:
> preferentially select inputs with only that asset

That's actually a good idea. And it shouldn't be too hard to implement.